internal/handlers/logs.go

```diff
@@ -135,13 +139,98 @@ func (p *Log) PatchLog(ctx *fiber.Ctx) error {
 func (p *Log) DeleteLog(ctx *fiber.Ctx) error {
 	var log models.Log
 
-	if err := p.db.Delete(&log, ctx.Params("id")).Error; err != nil {
-		if errors.Is(err, gorm.ErrRecordNotFound) {
-			return common.Error(fiber.StatusNotFound, "can't delete Log", "Log was not found")
-		}
+	result := p.db.Delete(&log, "id = ?", ctx.Params("id"))
 
+	if result.Error != nil {
 		return common.Error(fiber.StatusInternalServerError, "can't delete Log", "db error")
 	}
 
+	if result.RowsAffected == 0 {
+		return common.Error(fiber.StatusNotFound, "can't delete Log", "Log was not found")
+	}
+
 	return ctx.SendStatus(fiber.StatusNoContent)
 }
+
+// GetSoftwareLogs gets the logs associated to a Software with the given ID and returns any error encountered.
+func (p *Log) GetSoftwareLogs(ctx *fiber.Ctx) error {
+	var logs []models.Log
+
+	software := models.Software{}
+
+	if err := p.db.First(&software, "id = ?", ctx.Params("id")).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return common.Error(fiber.StatusNotFound, "can't get Software", "Software was not found")
+		}
+
+		return common.Error(
+			fiber.StatusInternalServerError,
+			"can't get Software",
+			fiber.ErrInternalServerError.Message,
+		)
+	}
+
+	stmt := p.db.
+		Where(map[string]interface{}{"entity_type": models.Software{}.TableName()}).
```

Comment:
@LorenzoS92 updated

Still ugly as sin, but at least it's not an hardcoded string